src/fabric/src/fabric2_fdb.erl

```diff
@@ -177,10 +177,13 @@ create(#{} = Db0, Options) ->
         layer_prefix := LayerPrefix
     } = Db = ensure_current(Db0, false),
 
-    % Eventually DbPrefix will be HCA allocated. For now
-    % we're just using the DbName so that debugging is easier.
     DbKey = erlfdb_tuple:pack({?ALL_DBS, DbName}, LayerPrefix),
-    DbPrefix = erlfdb_tuple:pack({?DBS, DbName}, LayerPrefix),
+    DefDbPref = ?DEFAULT_DB_PREFIX,
+    AllDbPrefix = erlfdb_util:get(Options, db_prefix, DefDbPref),
+    DbId = erlfdb_tuple:pack({AllDbPrefix}, AllDbPrefix),
+    DbPrefixAllocator = erlfdb_hca:create(erlfdb_tuple:pack({DbId}, <<"hca">>)),
+    AllocPrefix = erlfdb_hca:allocate(DbPrefixAllocator, Tx),
+    DbPrefix = erlfdb_tuple:pack({?DBS, AllocPrefix}, LayerPrefix),
```

Comment:
I wonder we could just keep the HCA counters right under the the `LayerPrefix?

As in:

```
DbPrefixAlloc = erlfdb_hca:create(erlfdb_tuple:pack({?DB_HCA}, LayerPrefix))
```

And then define `DB_HCA` to be in fabric2.hrl alongside `ALL_DBS`. 

```
...
-define(ALL_DBS, 1).
-define(DB_HCA, 2).
```

Then we don't even need the `DEFAULT_DB_PREFIX` since we'd always be creating these instances under `LayerPrefix` as well. In other words, they won't clash with FDB directory nodes.

Would that work?

Reply:
That's the correct approach. The directory prefix thing is just defined there because its for layer coordination. We just need to create our own subspace like `DB_HCA` as mentioned and all HCA operations just take that key as a parameter if memory serves.

Reply:
thanks @nickva and @davisp. I simplified creation of HCA according to your suggestion.